File: PaED62__perl/PaED62.pl
#!/usr/bin/perl -l

use strict;
use warnings;

sub hash
{
	my @input = split(//, $_[0]);
	my @output = split(//, "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz");

	for (my $j = 0; $j < scalar(@input); $j++)
	{
		for (my $i = 0; $i < scalar(@output); $i++)
		{
			$output[$i] = chr(((ord($output[$i]) * ($i + 1)) + (ord($input[$j]) * ($j + 1))) % (126 - 33) + 33);
		}
	}

	return join("", @output);
}

print(hash("Hello, world!"));
print(hash("Hello, world! "));

<STDIN>;

File: PaED62__py/PaED62.py
def hash(s):
	sInput = list(s);
	sOutput = list("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz");

	for j in range(len(sInput)):
		for i in range(len(sOutput)):
			sOutput[i] = chr(((ord(sOutput[i]) * (i + 1)) + (ord(sInput[j]) * (j + 1))) % (126 - 33) + 33);

	return "".join(sOutput);

print(hash("Hello, world!"));
print(hash("Hello, world! "));